package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs

```diff
@@ -489,7 +489,7 @@ public string usdFullPath
     /// </summary>
     public void DestroyAllImportedObjects() {
       foreach (var src in GetComponentsInChildren<UsdPrimSource>(includeInactive: true)) {
-        if (src) {
+        if (src && src.gameObject != this.gameObject) {
```

Comment:
Could you precise why you added this check? `this.gameObject` isn't part of "AllImportedObjects"?

Reply:
Good call -- done

Reply:
I added a comment in the code, but to explain here: the current game object is the UsdAsset root object. If it is removed, then the entire imported object will be gone. The current thought is to remove all objects EXCEPT the root, so that you can have a workflow where you leave stubs of USD Assets in the scene and import them only as needed.